### PR TITLE
[21.02] various script backports

### PR DESCRIPTION
**Pull request by ynezz**
Backports following script related backports:

 * scripts/download.pl: fix downloads with wget
 * scripts/download.pl: silence can't exec curl warning
 * scripts: always check certificates
 * tools: remove xxd package
 * build: provide xxd -i with scripts/xxdi.pl
 * scripts: xxdi.pl: add xxd -i compat mode
 * scripts: xxdi.pl: remove File::Slurp dependency
 * scripts: add xxdi.pl

### CODE REVIEW

File: scripts/download.pl [hunk 1]
```diff
@@ -74,16 +76,22 @@ ($)
...
 
 	return $have_curl
-		? (qw(curl -f --connect-timeout 20 --retry 5 --location --insecure), shellwords($ENV{CURL_OPTIONS} || ''), $url)
-		: (qw(wget --tries=5 --timeout=20 --no-check-certificate --output-document=-), shellwords($ENV{WGET_OPTIONS} || ''), $url)
+		? (qw(curl -f --connect-timeout 20 --retry 5 --location),
+			$check_certificate ? () : '--insecure',
+			shellwords($ENV{CURL_OPTIONS} || ''),
+			$url)
+		: (qw(wget --tries=5 --timeout=20 --output-document=-),
+			$check_certificate ? () : '--no-check-certificate',
```
Comment (Ansuel) [inline scripts/download.pl:92]: wait the fix for wget got dropped ?

Comment (ynezz) [reply scripts/download.pl:92]: `''` -> `()` is the fix.

Comment (Ansuel) [reply scripts/download.pl:92]: Oh ok... sorry tought it was still the old patch with `qw()`

Comment (ynezz) [reply scripts/download.pl:92]: No problem, thanks for the review :)

**Review by neheb — APPROVED** [review]

**Review by hauke — APPROVED** [review]

**Comment by ynezz** [comment]
Thank you for the review, merged in c07c565ea6.

**Pull request closed without merging**